### PR TITLE
node: add backer loop

### PR DESCRIPTION
**Pull request by libotony**

### CODE REVIEW

File: cmd/thor/node/backer_loop.go [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+	var scope event.SubscriptionScope
+	defer scope.Close()
+
+	newProposalCh := make(chan *comm.NewProposalEvent)
+	scope.Track(n.comm.SubscribeProposal(newProposalCh))
+
+	newAcceptedCh := make(chan *comm.NewAcceptedEvent)
+	scope.Track(n.comm.SubscribeAccepted(newAcceptedCh))
+
+	unknownTicker := time.NewTicker(time.Duration(1) * time.Second)
```
Comment (qianbin) [inline cmd/thor/node/backer_loop.go:142]: n * time.Second is ok

File: cmd/thor/node/backer_loop.go [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+				}
+
+				if err := n.validateBackerSignature(ev.Signature, p, st); err != nil {
+					log.Debug("failed to validate backer signature", "err", err)
+					continue
+				}
+
+				n.comm.BroadcastAccepted(ev.Accepted)
+			} else {
+				unknownAccepted.Set(ev.Hash(), ev.Accepted)
```
Comment (qianbin) [inline cmd/thor/node/backer_loop.go:241]: should be validated here?

File: cmd/thor/node/backer_loop.go [hunk 3]
```diff
@@ -0,0 +1,427 @@
...
+				continue
+			}
+			seenProposal.Add(hash, struct{}{})
+
+			if parent.Header().ID() != p.ParentID {
+				unknownProposal.Set(hash, ev.Proposal)
+				continue
+			}
+
+			if st == nil || st.ParentID() != parent.Header().ID() {
```
Comment (qianbin) [inline cmd/thor/node/backer_loop.go:173]: it's better to maintain bestblock(parent) and st under unknownTicker only.

File: cmd/thor/node/backer_loop.go (hunk 2 above)
Comment (libotony) [reply cmd/thor/node/backer_loop.go:241]: The backer signature can not be validated without the proposal

**Pull request closed without merging**